draft-irtf-cfrg-hpke.md

```diff
@@ -1207,6 +1207,43 @@ across all labels in this document.
 The `0xFFFF` AEAD ID is reserved for applications which only use the Export
 interface; see {{hpke-export}} for more details.
 
+# API Considerations {#api-considerations}
+
+The high-level HPKE APIs specified in this document are all fallible.
+For example, `Decap()` can fail if the encapsulated key `enc` is invalid,
+and `Open()` may fail if ciphertext decryption fails. The explicit errors
+generated throughout this specification, along with the conditions that
+lead to each error, are as follows:
+
+- `ValidationError`: KEM input or output validation failure; {{dhkem}}.
+- `DeserializeError`: Public or private key deserialization failure; {{base-crypto}}.
+- `DecapError`: `Decap()` failure; {{base-crypto}}.
+- `OpenError`: Context AEAD `Open()` failure; {{base-crypto}} and {{hpke-dem}}.
+- `MessageLimitReachedError`: Context AEAD sequence number overflow; {{base-crypto}} and {{hpke-dem}}.
+
+Implicit errors may also occur. As an example, certain classes of failures,
+e.g., malformed recipient public keys, may not yield explicit errors.
+For example, for the DHKEM variant described in this specification,
+the `Encap()` algorithm may fail when given an invalid recipient public key.
```

Comment:
Sec 7.1.4 actually says "the sender MUST validate the recipient's public key pkR"

```suggestion
the `Encap()` algorithm must fail when given an invalid recipient public key.
```